<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  
  <title>usage of machiato classifier | MaChIAto</title>
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
  
  <meta name="keywords" content="Usage" />
  
  
  
  
  <meta name="description" content="Parameter list-m or –mode {str}: This parameter allows for the specification of the type of analysis: “CRISPResso” and “CRISPResso2” is allowed in the latest version -a or –amplicon_seq {str}: This pa">
<meta property="og:type" content="article">
<meta property="og:title" content="Usage of MaChIAto Classifier">
<meta property="og:url" content="https://github.com/2022/06/22/Usage-of-MaChIAto-Classifier/index.html">
<meta property="og:site_name" content="MaChIAto">
<meta property="og:description" content="Parameter list-m or –mode {str}: This parameter allows for the specification of the type of analysis: “CRISPResso” and “CRISPResso2” is allowed in the latest version -a or –amplicon_seq {str}: This pa">
<meta property="og:locale" content="en_US">
<meta property="article:published_time" content="2022-06-21T15:26:26.000Z">
<meta property="article:modified_time" content="2022-07-27T09:32:29.705Z">
<meta property="article:author" content="Kazuki Nakamae, Ph.D., Hiroshima Univ, Japan.">
<meta property="article:tag" content="Usage">
<meta name="twitter:card" content="summary">
  
    <link rel="alternate" href="/atom.xml" title="MaChIAto" type="application/atom+xml">
  

  

  <link rel="icon" href="/css/images/logo.png">
  <link rel="apple-touch-icon" href="/css/images/logo.png">
  
    <link href="//fonts.googleapis.com/css?family=Source+Code+Pro" rel="stylesheet" type="text/css">
  
  <link href="https://fonts.googleapis.com/css?family=Open+Sans|Montserrat:700" rel="stylesheet" type="text/css">
  <link href="https://fonts.googleapis.com/css?family=Roboto:400,300,300italic,400italic" rel="stylesheet" type="text/css">
  <link href="//netdna.bootstrapcdn.com/font-awesome/4.0.3/css/font-awesome.css" rel="stylesheet">
  <style type="text/css">
    @font-face{font-family:futura-pt; src:url("/css/fonts/FuturaPTBold.otf") format("woff");font-weight:500;font-style:normal;}
    @font-face{font-family:futura-pt-light; src:url("/css/fonts/FuturaPTBook.otf") format("woff");font-weight:lighter;font-style:normal;}
    @font-face{font-family:futura-pt-italic; src:url("/css/fonts/FuturaPTBookOblique.otf") format("woff");font-weight:400;font-style:italic;}
}

  </style>
  
<link rel="stylesheet" href="/css/style.css">


  
<script src="/js/jquery-3.1.1.min.js"></script>

  
<script src="/js/bootstrap.js"></script>


  <!-- Bootstrap core CSS -->
  <link rel="stylesheet" href="/css/bootstrap.css" >

  
    
<link rel="stylesheet" href="/css/dialog.css">

  

  

  
    <link rel="stylesheet" href="/css/header-post.css" >
  

  
  
  

<meta name="generator" content="Hexo 6.2.0"></head>



  <body data-spy="scroll" data-target="#toc" data-offset="50">


  
  <div id="container">
    <div id="wrap">
      
        <header>

    <div id="allheader" class="navbar navbar-default navbar-static-top" role="navigation">
        <div class="navbar-inner">
          
          <div class="container"> 
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
              <span class="sr-only">Toggle navigation</span>
              <span class="icon-bar"></span>
              <span class="icon-bar"></span>
              <span class="icon-bar"></span>
            </button>

            
              <a class="brand" style="
                 margin-top: 0px;"  
                href="#" data-toggle="modal" data-target="#myModal" >
                  <img width="124px" height="124px" alt="Hike News" src="/css/images/logo.png">
              </a>
            
            
            <div class="navbar-collapse collapse">
              <ul class="hnav navbar-nav">
                
                  <li> <a class="main-nav-link" href="/">Home</a> </li>
                
                  <li> <a class="main-nav-link" href="/archives">Archives</a> </li>
                
                  <li> <a class="main-nav-link" href="/download">download</a> </li>
                
                  <li> <a class="main-nav-link" href="/citation">citation</a> </li>
                
                  <li> <a class="main-nav-link" href="/about">About</a> </li>
                
                  <li> <a class="main-nav-link" href="/contactus">contact</a> </li>
                
                  <li><div id="search-form-wrap">

    <form class="search-form">
        <input type="text" class="ins-search-input search-form-input" placeholder="" />
        <button type="submit" class="search-form-submit"></button>
    </form>
    <div class="ins-search">
    <div class="ins-search-mask"></div>
    <div class="ins-search-container">
        <div class="ins-input-wrapper">
            <input type="text" class="ins-search-input" placeholder="Type something..." />
            <span class="ins-close ins-selectable"><i class="fa fa-times-circle"></i></span>
        </div>
        <div class="ins-section-wrapper">
            <div class="ins-section-container"></div>
        </div>
    </div>
</div>
<script>
(function (window) {
    var INSIGHT_CONFIG = {
        TRANSLATION: {
            POSTS: 'Posts',
            PAGES: 'Pages',
            CATEGORIES: 'Categories',
            TAGS: 'Tags',
            UNTITLED: '(Untitled)',
        },
        ROOT_URL: '/',
        CONTENT_URL: '/content.json',
    };
    window.INSIGHT_CONFIG = INSIGHT_CONFIG;
})(window);
</script>

<script src="/js/insight.js"></script>


</div></li>
            </div>
          </div>
                
      </div>
    </div>

</header>



      
            
      <div id="content" class="outer">
        
          <section id="main" style="float:none;"><article id="post-Usage-of-MaChIAto-Classifier" style="width: 75%; float:left;" class="article article-type-post" itemscope itemprop="blogPost" >
  <div id="articleInner" class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 class="thumb" class="article-title" itemprop="name">
      Usage of MaChIAto Classifier
    </h1>
  

      </header>
    
    <div class="article-meta">
      
	<a href="/2022/06/22/Usage-of-MaChIAto-Classifier/" class="article-date">
	  <time datetime="2022-06-21T15:26:26.000Z" itemprop="datePublished">2022-06-22</time>
	</a>

      
    <a class="article-category-link" href="/categories/Introduction/">Introduction</a>

      
	<a class="article-views">
	<span id="busuanzi_container_page_pv">
		PV:<span id="busuanzi_value_page_pv"></span>
	</span>
	</a>

      

    </div>
    <div class="article-entry" itemprop="articleBody">
      
        <h3 id="Parameter-list"><a href="#Parameter-list" class="headerlink" title="Parameter list"></a>Parameter list</h3><p><strong>-m</strong> or <strong>–mode</strong> {str}: This parameter allows for the specification of the type of analysis: “CRISPResso” and “CRISPResso2” is allowed in the latest version</p>
<p><strong>-a</strong> or <strong>–amplicon_seq</strong> {str}: This parameter allows the user to enter the amplicon sequence used for the CRISPResso. The length should be &gt;105bp due to the setting for the other parameter.</p>
<p><strong>-g</strong> or <strong>–guide_seq</strong> {str}: This parameter allows for the specification of the sgRNA sequence used for the CRISPResso. The length of the sequence should be 20nt without PAM. The MaChIAto convention is to depict the expected cleavage position using the value of the parameter three nt 3’ from the end of the guide.”, required&#x3D;True)</p>
<p><strong>-cf</strong> or <strong>–crispreeso_file</strong> {str}: This parameter allows for the specification of the “Alleles_frequency_table.txt” from CRISPResso. When this parameter is used, “CRISPResso” should be entered as -m parameter.’, (default: “.&#x2F;Alleles_frequency_table.txt”) (optional)</p>
<p><strong>-ccf</strong> or <strong>–crispreeso2_file</strong> {str}: This parameter allows for the specification of the “Alleles_frequency_table.zip” from CRISPResso2. When this parameter is used, “CRISPResso2” should be entered as -m parameter.’, (default: “.&#x2F;Alleles_frequency_table.zip”) (optional)</p>
<p><strong>-d</strong> or <strong>–donor_seq</strong> {str}:This parameter allows for the specification of the expected HDR amplicon used for the CRISPResso. The length of sequence should be &gt;12bp in knock-out&#x2F;knock-in analysis. In knock-out analysis, this parameter should not be entered, and then the value will be given fake parameter (“TTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTT”), and some internal settings are changed for knock-out analysis. However, the fake parameter will be poly-C|G|A if amplicon sequence contains poly-T sequence.’, (default: “”) (optional)</p>
<p><strong>-e</strong> or <strong>–expected_ki_amplicon_seq</strong> {str}: This parameter allows for the specification of the expected knock-in amplicon sequence used for the CRISPResso after HDR. The length of the sequence should be &gt;12bp. In knock-out analysis, this parameter should not be entered, and then the value will be given fake parameter including fake donor sequence, and some internal settings are changed for knock-out analysis.’, (default: “”) (optional)</p>
<p><strong>-o</strong> or <strong>–output_folder</strong> {str}: This parameter allows for the specification of the output directory to use for the analysis (default: current directory) (default: “.&#x2F;“) (optional)</p>
<p><strong>-lh</strong> or <strong>–length_left_homologyarm</strong> {int}: This parameter allows for the specification of the length of 5’ homology arm (default: 20). The length of the sequence should be &gt;17bp, and the flanking sequence with the homology arm needs the length of &gt;24bp in the expected amplicon sequence. In the prime editing analysis, the 5’ homology arm is considered as the RT template (default: 20) (optional)</p>
<p><strong>-rh</strong> or <strong>–length_right_homologyarm</strong> {int}: This parameter allows for the specification of the length of 3’ homology arm (default: 20). The length of the sequence should be &gt;17bp, and the flanking sequence with the homology arm needs the length of &gt;24bp in the expected amplicon sequence. In the prime editing analysis, the 3’ homology arm is considered as the prime binding site (default: 20) (optional)</p>
<p><strong>-cn</strong> or <strong>–location_comp_nick</strong> {int}: This parameter allows for the specification of the complementary strand nick location [3prime direction is +] (default: 90). This parameter is used in the prime editing and should be over the homology arm’s length to which the nickase is adjacent’ (default: 90) (optional)</p>
<p><strong>-n</strong> or <strong>–name</strong>  : This parameter allows for the specification of the name, which will be included output directory (default: “untitled-X”). If MaChIAto Analyzer and MaChIAto Reviewer will be used in the following analysis, the value should be “{target_name}-{sample label }” (e.g., DBF4B-C) and underbar “_” should not be used’, (default: “untitled-X”) (optional)</p>
<p><strong>–primeediting_analysis</strong> : Re-classify the data as prime editing analysis. This option forces the setting to change for prime editing analysis.’ (optional)</p>
<h4 id="Advanced-parameter-for-developers"><a href="#Advanced-parameter-for-developers" class="headerlink" title="Advanced parameter (for developers)"></a>Advanced parameter (for developers)</h4><p><strong>–force_knockout_analysis</strong> : Usually, MaChIAto re-classify the data as knock-in analysis. This option forces the setting to change for knock-out analysis. Under this mode, the length of indicator on the knock-in donor is the maximum value, and the threshold value for alignment of the knock-in sequence is 1.0. If this parameter is not entered, MaChIAto can automatically set up this mode by finding some characteristics of the knock-out sample. For example, MaChIAto checks that there is no donor sequence or expected knock-in sequence as input, and there are less three kinds HDR variants among input data. (optional)</p>
<p><strong>–skip_optimization</strong> : Usually, MaChIAto runs Bayesian optimization for finding the optimized setting. This option allows MaChIAto to skip the process of optimization. The option is made for debugging. So, the option should not be used in the usual analysis. However, if the optimization process disturbs an accurate analysis, this option might be useful. (optional)</p>
<p><strong>–copy_optimization</strong> : Usually, MaChIAto runs Bayesian optimization for finding the optimized setting. This option allows MaChIAto to use the provided optimization data instead of the optimization. If you analyze NGS data with the setting of previously analyzed data, this option is useful. Especially in substitution editing, we can recommend applying the option to the negative&#x2F;positive control sample. However, you should understand that a classification error might frequently occur. (optional)</p>
<p><strong>–provided_optimization_file</strong> : This parameter allows for the specification of the “MaChIAto_optimized_param.csv” from the analyzed MaChIAto folder. When this parameter is used, –copy_optimization parameter is required.’, (default: “.&#x2F;MaChIAto_optimized_param.csv”) (optional)</p>
<h4 id="NOTE"><a href="#NOTE" class="headerlink" title="NOTE:"></a>NOTE:</h4><p>You can skip the optimization process using the “–skip_optimization” flag. However, we recommend you not to use the option unless you check the command and perform debugging. For example, suppose executing MaChIAto Classifier with poor optimization cause missing the many reads that include expected edits. Moreover, if the indicator is accidentally located in a spontaneous mutation point, The Bit-by-Bit Alignment might misrecognize the editing read. The optimization can avoid the problem. Thus, it would be best if you turned on the optimization for finding the bona fide editing reads. </p>
<h3 id="Template-command"><a href="#Template-command" class="headerlink" title="Template command"></a>Template command</h3><h5 id="Case-of-knock-out"><a href="#Case-of-knock-out" class="headerlink" title="Case of knock-out"></a>Case of knock-out</h5><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">python MaChIAto/MaChIAto.py \</span><br><span class="line">-m CRISPResso2 \</span><br><span class="line">-ccf (Alleles_frequency_table.zip of the CRISPResso2 output) \</span><br><span class="line">-o (output directory) \</span><br><span class="line">-a (wt amplicon sequence) \</span><br><span class="line">-g (protospaser sequence) \</span><br><span class="line">-n (sample name)-(label name);</span><br></pre></td></tr></table></figure>

<h5 id="Case-of-homology-based-knock-in"><a href="#Case-of-homology-based-knock-in" class="headerlink" title="Case of homology-based knock-in"></a>Case of homology-based knock-in</h5><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">python MaChIAto/MaChIAto.py \</span><br><span class="line">-m CRISPResso2 \</span><br><span class="line">-ccf (Alleles_frequency_table.zip of the CRISPResso2 output) \</span><br><span class="line">-o (output directory) \</span><br><span class="line">-a (wt amplicon sequence) \</span><br><span class="line">-g (protospaser sequence) \</span><br><span class="line">-e (expected editing amplicon seqeunce) \</span><br><span class="line">-d (donor insert sequence) \</span><br><span class="line">-lh (length of 5\<span class="string">&#x27; homology arm) \</span></span><br><span class="line"><span class="string">-rh (length of 3\&#x27;</span> homology arm) \</span><br><span class="line">-n (sample name)-(label name);</span><br></pre></td></tr></table></figure>

<h5 id="Case-of-Prime-Editing-Substitution-x2F-Deletion-editing"><a href="#Case-of-Prime-Editing-Substitution-x2F-Deletion-editing" class="headerlink" title="Case of Prime Editing (Substitution&#x2F;Deletion editing)"></a>Case of Prime Editing (Substitution&#x2F;Deletion editing)</h5><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line">python MaChIAto/MaChIAto.py \</span><br><span class="line">-m CRISPResso2 \</span><br><span class="line">-ccf (Alleles_frequency_table.zip of the CRISPResso2 output) \</span><br><span class="line">-o (output directory) \</span><br><span class="line">-a (wt amplicon sequence) \</span><br><span class="line">-g (protospaser sequence) \</span><br><span class="line">-e (expected editing amplicon seqeunce) \</span><br><span class="line">-lh (length of prime binding site) \</span><br><span class="line">-rh (length of RT template) \</span><br><span class="line">-cn (distance between two nick sites) \</span><br><span class="line">--primeediting_analysis \</span><br><span class="line">-n (sample name)-(label name);</span><br></pre></td></tr></table></figure>

<h5 id="Case-of-Prime-Editing-Insertion-editing"><a href="#Case-of-Prime-Editing-Insertion-editing" class="headerlink" title="Case of Prime Editing (Insertion editing)"></a>Case of Prime Editing (Insertion editing)</h5><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line">python MaChIAto/MaChIAto.py \</span><br><span class="line">-m CRISPResso2 \</span><br><span class="line">-ccf (Alleles_frequency_table.zip of the CRISPResso2 output) \</span><br><span class="line">-o (output directory) \</span><br><span class="line">-a (wt amplicon sequence) \</span><br><span class="line">-g (protospaser sequence) \</span><br><span class="line">-e (expected editing amplicon seqeunce) \</span><br><span class="line">-d (donor insert sequence) \</span><br><span class="line">-lh (length of prime binding site) \</span><br><span class="line">-rh (length of RT template) \</span><br><span class="line">-cn (distance between two nick sites) \</span><br><span class="line">--primeediting_analysis \</span><br><span class="line">-n (sample name)-(label name);</span><br></pre></td></tr></table></figure>

<p>If you want to use the result of CRISPResso version1, the -ccf should be replased into -cf.<br>Example for CRISPResso version1</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">python MaChIAto/MaChIAto.py \</span><br><span class="line">-m CRISPResso \</span><br><span class="line">-cf (Alleles_frequency_table.txt of the CRISPResso output) \</span><br><span class="line">-o (output directory) \</span><br><span class="line">-a (wt amplicon sequence) \</span><br><span class="line">-g (protospaser sequence) \</span><br><span class="line">-n (sample name)-(label name);</span><br></pre></td></tr></table></figure>

<p>*The <strong>sample name</strong> and <strong>label name</strong> can be <strong>arbitrary</strong>. If you analyze the multiple experiment (e.g. knock-out and knock-in), the name should be different from others.</p>
<h3 id="The-example-of-the-command"><a href="#The-example-of-the-command" class="headerlink" title="The example of the command"></a>The example of the command</h3><p>Here is example of the command using docker.</p>
<p>If you don’t have the docker image for MaChIAto Classifier, please go the Preparation page: <a target="_blank" rel="noopener" href="https://machiatopage.github.io/2100/01/01/Preparation/">https://machiatopage.github.io/2100/01/01/Preparation/</a></p>
<p>The input file can be downloaded from the following link:<br><a target="_blank" rel="noopener" href="https://sourceforge.net/projects/machiato-example-files/files/example_CRISPResso2_output.zip/download">https://sourceforge.net/projects/machiato-example-files/files/example_CRISPResso2_output.zip/download</a></p>
<p>You can check the output file of the below examples following the following link:<br><a target="_blank" rel="noopener" href="https://sourceforge.net/projects/machiato-example-files/files/example_MaChIAtoClassifier_docker_output.zip/download">https://sourceforge.net/projects/machiato-example-files/files/example_MaChIAtoClassifier_docker_output.zip/download</a></p>
<h5 id="Case-of-knock-out-1"><a href="#Case-of-knock-out-1" class="headerlink" title="Case of knock-out"></a>Case of knock-out</h5><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">docker run --rm -v `pwd`:/DATA -w /DATA -i kazukinakamae/machiato:latest python ../MaChIAto.py \</span><br><span class="line">-m CRISPResso2 \</span><br><span class="line">-ccf example_CRISPResso2_output/CRISPResso_on_AAVS1-KO/Alleles_frequency_table.zip \</span><br><span class="line">-o example_MaChIAtoClassifier_docker_output \</span><br><span class="line">-a caggggtccgagagctcagctagtcttcttcctccaacccgggcccctatgtccacttcaggacagcatgtttgctgcctccagggatcctgtgtccccgagctgggaccaccttatattcccagggccggttaatgtggctctggttctgggtacttttatctgtcccctccaccccacagtggggccactagggacaggattggtgacagaaaagccccatccttaggcctcctccttcctagtctcctgatattgggtctaacccccacctcctgttaggcagattccttatctggtgacacacccccatttcctggagccatctctct \</span><br><span class="line">-g accccacagtggggccacta \</span><br><span class="line">-n AAVS1-KO;</span><br></pre></td></tr></table></figure>

<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">docker run --rm -v `pwd`:/DATA -w /DATA -i kazukinakamae/machiato:latest python ../MaChIAto.py \</span><br><span class="line">-m CRISPResso2 \</span><br><span class="line">-ccf example_CRISPResso2_output/CRISPResso_on_CD40LG-KO/Alleles_frequency_table.zip \</span><br><span class="line">-o example_MaChIAtoClassifier_docker_output \</span><br><span class="line">-a tgtagaggcagatatcattagctgtattctccttccgaatgacatttatcatatccttgttattccaaaatagatagaagatgaaaggaatcttcatgaagattttgtattcatgaaaacgatacagagatgcaacacaggagaaagatccttatccttactgaactgtgaggagattaaaagccagtttgaaggctttgtgaaggtaagcagcttaattactggtaaaagtgtcattgaaatattttactacatttgctagatcgggaaactgacaatgccaatgtttaaagattggttatagacacagacacacagacacacacacacatat \</span><br><span class="line">-g CTTATCCTTACTGAACTGTG \</span><br><span class="line">-n CD40LG-KO;</span><br></pre></td></tr></table></figure>

<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">docker run --rm -v `pwd`:/DATA -w /DATA -i kazukinakamae/machiato:latest python ../MaChIAto.py \</span><br><span class="line">-m CRISPResso2 \</span><br><span class="line">-ccf example_CRISPResso2_output/CRISPResso_on_DBF4B-KO/Alleles_frequency_table.zip \</span><br><span class="line">-o example_MaChIAtoClassifier_docker_output \</span><br><span class="line">-a tttctgagcaaagaagtaagttacatcgtgtccagccgcagagaagtaaaggcagagagcagtgggaaaagccatagaggctgccctagccctagccccagtgaggtcagagtggaaacatcggccatggttgatccaaaaggcagccaccccaggccttcacggaaacccgttgactcggtaagaacctcatgtaggaaaggtatgctgtgtaaacaaaggaagtaggctttggtgagccttttaaggcagtctctggttttaatttcatctggattaatttaatctctcatttgaattcagggactgagaatattttatcatagctctg \</span><br><span class="line">-g TTCACGGAAACCCGTTGACT \</span><br><span class="line">-n DBF4B-KO;</span><br></pre></td></tr></table></figure>

<h5 id="Case-of-homology-based-knock-in-1"><a href="#Case-of-homology-based-knock-in-1" class="headerlink" title="Case of homology-based knock-in"></a>Case of homology-based knock-in</h5><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">docker run --rm -v `pwd`:/DATA -w /DATA -i kazukinakamae/machiato:latest python ../MaChIAto.py \</span><br><span class="line">-m CRISPResso2 \</span><br><span class="line">-ccf example_CRISPResso2_output/CRISPResso_on_AAVS1-KI/Alleles_frequency_table.zip \</span><br><span class="line">-o example_MaChIAtoClassifier_docker_output \</span><br><span class="line">-a caggggtccgagagctcagctagtcttcttcctccaacccgggcccctatgtccacttcaggacagcatgtttgctgcctccagggatcctgtgtccccgagctgggaccaccttatattcccagggccggttaatgtggctctggttctgggtacttttatctgtcccctccaccccacagtggggccactagggacaggattggtgacagaaaagccccatccttaggcctcctccttcctagtctcctgatattgggtctaacccccacctcctgttaggcagattccttatctggtgacacacccccatttcctggagccatctctct \</span><br><span class="line">-g accccacagtggggccacta \</span><br><span class="line">-e caggggtccgagagctcagctagtcttcttcctccaacccgggcccctatgtccacttcaggacagcatgtttgctgcctccagggatcctgtgtccccgagctgggaccaccttatattcccagggccggttaatgtggctctggttctGGGTACTTTTATCTGTCCCCTCCACCCCACAGTGGGGCCAGACTATAAGGACCACGACGGAGACTACAAGGATCATGATATTGATTACAAAGACGATGACGATAAGCTAGGGACAGGATTGGTGACAGAAAAGCCCCATCCTTAGGcctcctccttcctagtctcctgatattgggtctaacccccacctcctgttaggcagattccttatctggtgacacacccccatttcctggagccatctctct \</span><br><span class="line">-d GACTATAAGGACCACGACGGAGACTACAAGGATCATGATATTGATTACAAAGACGATGACGATAAG \</span><br><span class="line">-lh 40 -rh 40 -n AAVS1-KI;</span><br></pre></td></tr></table></figure>

<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">docker run --rm -v `pwd`:/DATA -w /DATA -i kazukinakamae/machiato:latest python ../MaChIAto.py \</span><br><span class="line">-m CRISPResso2 \</span><br><span class="line">-ccf example_CRISPResso2_output/CRISPResso_on_CD40LG-KI/Alleles_frequency_table.zip \</span><br><span class="line">-o example_MaChIAtoClassifier_docker_output \</span><br><span class="line">-a tgtagaggcagatatcattagctgtattctccttccgaatgacatttatcatatccttgttattccaaaatagatagaagatgaaaggaatcttcatgaagattttgtattcatgaaaacgatacagagatgcaacacaggagaaagatccttatccttactgaactgtgaggagattaaaagccagtttgaaggctttgtgaaggtaagcagcttaattactggtaaaagtgtcattgaaatattttactacatttgctagatcgggaaactgacaatgccaatgtttaaagattggttatagacacagacacacagacacacacacacatat \</span><br><span class="line">-g CTTATCCTTACTGAACTGTG \</span><br><span class="line">-e tgtagaggcagatatcattagctgtattctccttccgaatgacatttatcatatccttgttattccaaaatagatagaagatgaaaggaatcttcatgaagattttgtattcatgaaaacgatacagagatgcaacacaggagaaagatccttatccttactgaactgACTATAAGGACCACGACGGAGACTACAAGGATCATGATATTGATTACAAAGACGATGACGATAAGGtgaggagattaaaagccagtttgaaggctttgtgaaggtaagcagcttaattactggtaaaagtgtcattgaaatattttactacatttgctagatcgggaaactgacaatgccaatgtttaaagattggttatagacacagacacacagacacacacacacatat \</span><br><span class="line">-d gACTATAAGGACCACGACGGAGACTACAAGGATCATGATATTGATTACAAAGACGATGACGATAAG \</span><br><span class="line">-lh 40 -rh 40 -n CD40LG-KI;</span><br></pre></td></tr></table></figure>

<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">docker run --rm -v `pwd`:/DATA -w /DATA -i kazukinakamae/machiato:latest python ../MaChIAto.py \</span><br><span class="line">-m CRISPResso2 \</span><br><span class="line">-ccf example_CRISPResso2_output/CRISPResso_on_DBF4B-KI/Alleles_frequency_table.zip \</span><br><span class="line">-o example_MaChIAtoClassifier_docker_output \</span><br><span class="line">-a tttctgagcaaagaagtaagttacatcgtgtccagccgcagagaagtaaaggcagagagcagtgggaaaagccatagaggctgccctagccctagccccagtgaggtcagagtggaaacatcggccatggttgatccaaaaggcagccaccccaggccttcacggaaacccgttgactcggtaagaacctcatgtaggaaaggtatgctgtgtaaacaaaggaagtaggctttggtgagccttttaaggcagtctctggttttaatttcatctggattaatttaatctctcatttgaattcagggactgagaatattttatcatagctctg \</span><br><span class="line">-g TTCACGGAAACCCGTTGACT \</span><br><span class="line">-e tttctgagcaaagaagtaagttacatcgtgtccagccgcagagaagtaaaggcagagagcagtgggaaaagccatagaggctgccctagccctagccccagtgaggtcagagtggaaacatcggccatggttgatCCAAAAGGCAGCCACCCCAGGCCTTCACGGAAACCCGTTGGACTATAAGGACCACGACGGAGACTACAAGGATCATGATATTGATTACAAAGACGATGACGATAAGACTCGGTAAGAACCTCATGTAGGAAAGGTATGCTGTGTAAacaaaggaagtaggctttggtgagccttttaaggcagtctctggttttaatttcatctggattaatttaatctctcatttgaattcagggactgagaatattttatcatagctctg \</span><br><span class="line">-d GACTATAAGGACCACGACGGAGACTACAAGGATCATGATATTGATTACAAAGACGATGACGATAAG \</span><br><span class="line">-lh 40 -rh 40 -n DBF4B-KI;</span><br></pre></td></tr></table></figure>

<h5 id="Case-of-untreated-sample-for-negative-control-Optional"><a href="#Case-of-untreated-sample-for-negative-control-Optional" class="headerlink" title="Case of untreated sample for negative control (Optional)"></a>Case of untreated sample for negative control (Optional)</h5><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">docker run --rm -v `pwd`:/DATA -w /DATA -i kazukinakamae/machiato:latest python ../MaChIAto.py \</span><br><span class="line">-m CRISPResso2 \</span><br><span class="line">-ccf example_CRISPResso2_output/CRISPResso_on_AAVS1-WT/Alleles_frequency_table.zip \</span><br><span class="line">-o example_MaChIAtoClassifier_docker_output \</span><br><span class="line">-a caggggtccgagagctcagctagtcttcttcctccaacccgggcccctatgtccacttcaggacagcatgtttgctgcctccagggatcctgtgtccccgagctgggaccaccttatattcccagggccggttaatgtggctctggttctgggtacttttatctgtcccctccaccccacagtggggccactagggacaggattggtgacagaaaagccccatccttaggcctcctccttcctagtctcctgatattgggtctaacccccacctcctgttaggcagattccttatctggtgacacacccccatttcctggagccatctctct \</span><br><span class="line">-g accccacagtggggccacta \</span><br><span class="line">-n AAVS1-WT;</span><br></pre></td></tr></table></figure>

<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">docker run --rm -v `pwd`:/DATA -w /DATA -i kazukinakamae/machiato:latest python ../MaChIAto.py \</span><br><span class="line">-m CRISPResso2 \</span><br><span class="line">-ccf example_CRISPResso2_output/CRISPResso_on_CD40LG-WT/Alleles_frequency_table.zip \</span><br><span class="line">-o example_MaChIAtoClassifier_docker_output \</span><br><span class="line">-a tgtagaggcagatatcattagctgtattctccttccgaatgacatttatcatatccttgttattccaaaatagatagaagatgaaaggaatcttcatgaagattttgtattcatgaaaacgatacagagatgcaacacaggagaaagatccttatccttactgaactgtgaggagattaaaagccagtttgaaggctttgtgaaggtaagcagcttaattactggtaaaagtgtcattgaaatattttactacatttgctagatcgggaaactgacaatgccaatgtttaaagattggttatagacacagacacacagacacacacacacatat \</span><br><span class="line">-g CTTATCCTTACTGAACTGTG \</span><br><span class="line">-n CD40LG-WT;</span><br></pre></td></tr></table></figure>

<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">docker run --rm -v `pwd`:/DATA -w /DATA -i kazukinakamae/machiato:latest python ../MaChIAto.py \</span><br><span class="line">-m CRISPResso2 \</span><br><span class="line">-ccf example_CRISPResso2_output/CRISPResso_on_DBF4B-WT/Alleles_frequency_table.zip \</span><br><span class="line">-o example_MaChIAtoClassifier_docker_output \</span><br><span class="line">-a tttctgagcaaagaagtaagttacatcgtgtccagccgcagagaagtaaaggcagagagcagtgggaaaagccatagaggctgccctagccctagccccagtgaggtcagagtggaaacatcggccatggttgatccaaaaggcagccaccccaggccttcacggaaacccgttgactcggtaagaacctcatgtaggaaaggtatgctgtgtaaacaaaggaagtaggctttggtgagccttttaaggcagtctctggttttaatttcatctggattaatttaatctctcatttgaattcagggactgagaatattttatcatagctctg \</span><br><span class="line">-g TTCACGGAAACCCGTTGACT \</span><br><span class="line">-n DBF4B-WT;</span><br></pre></td></tr></table></figure>

<h5 id="Case-of-homology-based-knock-in-that-is-the-different-condition-from-the-above-method-Optional"><a href="#Case-of-homology-based-knock-in-that-is-the-different-condition-from-the-above-method-Optional" class="headerlink" title="Case of homology-based knock-in that is the different condition from the above method (Optional)"></a>Case of homology-based knock-in that is the different condition from the above method (Optional)</h5><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">docker run --rm -v `pwd`:/DATA -w /DATA -i kazukinakamae/machiato:latest python ../MaChIAto.py \</span><br><span class="line">-m CRISPResso2 \</span><br><span class="line">-ccf example_CRISPResso2_output/CRISPResso_on_AAVS1-KI2/Alleles_frequency_table.zip \</span><br><span class="line">-o example_MaChIAtoClassifier_docker_output \</span><br><span class="line">-a caggggtccgagagctcagctagtcttcttcctccaacccgggcccctatgtccacttcaggacagcatgtttgctgcctccagggatcctgtgtccccgagctgggaccaccttatattcccagggccggttaatgtggctctggttctgggtacttttatctgtcccctccaccccacagtggggccactagggacaggattggtgacagaaaagccccatccttaggcctcctccttcctagtctcctgatattgggtctaacccccacctcctgttaggcagattccttatctggtgacacacccccatttcctggagccatctctct \</span><br><span class="line">-g accccacagtggggccacta \</span><br><span class="line">-e caggggtccgagagctcagctagtcttcttcctccaacccgggcccctatgtccacttcaggacagcatgtttgctgcctccagggatcctgtgtccccgagctgggaccaccttatattcccagggccggttaatgtggctctggttctGGGTACTTTTATCTGTCCCCTCCACCCCACAGTGGGGCCAGACTATAAGGACCACGACGGAGACTACAAGGATCATGATATTGATTACAAAGACGATGACGATAAGCTAGGGACAGGATTGGTGACAGAAAAGCCCCATCCTTAGGcctcctccttcctagtctcctgatattgggtctaacccccacctcctgttaggcagattccttatctggtgacacacccccatttcctggagccatctctct \</span><br><span class="line">-d GACTATAAGGACCACGACGGAGACTACAAGGATCATGATATTGATTACAAAGACGATGACGATAAG \</span><br><span class="line">-lh 40 -rh 40 -n AAVS1-KI2;</span><br></pre></td></tr></table></figure>

<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">docker run --rm -v `pwd`:/DATA -w /DATA -i kazukinakamae/machiato:latest python ../MaChIAto.py \</span><br><span class="line">-m CRISPResso2 \</span><br><span class="line">-ccf example_CRISPResso2_output/CRISPResso_on_CD40LG-KI2/Alleles_frequency_table.zip \</span><br><span class="line">-o example_MaChIAtoClassifier_docker_output \</span><br><span class="line">-a tgtagaggcagatatcattagctgtattctccttccgaatgacatttatcatatccttgttattccaaaatagatagaagatgaaaggaatcttcatgaagattttgtattcatgaaaacgatacagagatgcaacacaggagaaagatccttatccttactgaactgtgaggagattaaaagccagtttgaaggctttgtgaaggtaagcagcttaattactggtaaaagtgtcattgaaatattttactacatttgctagatcgggaaactgacaatgccaatgtttaaagattggttatagacacagacacacagacacacacacacatat \</span><br><span class="line">-g CTTATCCTTACTGAACTGTG \</span><br><span class="line">-e tgtagaggcagatatcattagctgtattctccttccgaatgacatttatcatatccttgttattccaaaatagatagaagatgaaaggaatcttcatgaagattttgtattcatgaaaacgatacagagatgcaacacaggagaaagatccttatccttactgaactgACTATAAGGACCACGACGGAGACTACAAGGATCATGATATTGATTACAAAGACGATGACGATAAGGtgaggagattaaaagccagtttgaaggctttgtgaaggtaagcagcttaattactggtaaaagtgtcattgaaatattttactacatttgctagatcgggaaactgacaatgccaatgtttaaagattggttatagacacagacacacagacacacacacacatat \</span><br><span class="line">-d gACTATAAGGACCACGACGGAGACTACAAGGATCATGATATTGATTACAAAGACGATGACGATAAG \</span><br><span class="line">-lh 40 -rh 40 -n CD40LG-KI2;</span><br></pre></td></tr></table></figure>

<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">docker run --rm -v `pwd`:/DATA -w /DATA -i kazukinakamae/machiato:latest python ../MaChIAto.py \</span><br><span class="line">-m CRISPResso2 \</span><br><span class="line">-ccf example_CRISPResso2_output/CRISPResso_on_DBF4B-KI2/Alleles_frequency_table.zip \</span><br><span class="line">-o example_MaChIAtoClassifier_docker_output \</span><br><span class="line">-a tttctgagcaaagaagtaagttacatcgtgtccagccgcagagaagtaaaggcagagagcagtgggaaaagccatagaggctgccctagccctagccccagtgaggtcagagtggaaacatcggccatggttgatccaaaaggcagccaccccaggccttcacggaaacccgttgactcggtaagaacctcatgtaggaaaggtatgctgtgtaaacaaaggaagtaggctttggtgagccttttaaggcagtctctggttttaatttcatctggattaatttaatctctcatttgaattcagggactgagaatattttatcatagctctg \</span><br><span class="line">-g TTCACGGAAACCCGTTGACT \</span><br><span class="line">-e tttctgagcaaagaagtaagttacatcgtgtccagccgcagagaagtaaaggcagagagcagtgggaaaagccatagaggctgccctagccctagccccagtgaggtcagagtggaaacatcggccatggttgatCCAAAAGGCAGCCACCCCAGGCCTTCACGGAAACCCGTTGGACTATAAGGACCACGACGGAGACTACAAGGATCATGATATTGATTACAAAGACGATGACGATAAGACTCGGTAAGAACCTCATGTAGGAAAGGTATGCTGTGTAAacaaaggaagtaggctttggtgagccttttaaggcagtctctggttttaatttcatctggattaatttaatctctcatttgaattcagggactgagaatattttatcatagctctg \</span><br><span class="line">-d GACTATAAGGACCACGACGGAGACTACAAGGATCATGATATTGATTACAAAGACGATGACGATAAG \</span><br><span class="line">-lh 40 -rh 40 -n DBF4B-KI2;</span><br></pre></td></tr></table></figure>


      
    </div>
    <footer class="article-footer">
      
      
      <div>
        <ul class="post-copyright">
          <li class="post-copyright-author">
          <strong>Post author:  </strong>Kazuki Nakamae, Ph.D., Hiroshima Univ, Japan.</a>
          </li>
          <li class="post-copyright-link">
          <strong>Post link:  </strong>
          <a href="/2022/06/22/Usage-of-MaChIAto-Classifier/" target="_blank" title="Usage of MaChIAto Classifier">https://github.com/2022/06/22/Usage-of-MaChIAto-Classifier/</a>
          </li>
          <li class="post-copyright-license">
            <strong>Copyright Notice:   </strong>
            All articles in this blog are licensed under <a rel="license" href="https://creativecommons.org/licenses/by-nc-nd/4.0/" target="_blank" title="Attribution-NonCommercial-NoDerivatives 4.0 International (CC BY-NC-ND 4.0)">CC BY-NC-ND 4.0</a>
            unless stating additionally.
          </li>
         
        </ul>
<div>

      
      
      
        
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/Usage/" rel="tag">Usage</a></li></ul>

      

    </footer>
  </div>
  
    
<nav id="article-nav">
  
    <a href="/2022/06/22/Output-of-MaChIAto-Classifier/" id="article-nav-newer" class="article-nav-link-wrap">
      <strong class="article-nav-caption">Newer</strong>
      <div class="article-nav-title">
        
          Output of MaChIAto Classifier
        
      </div>
    </a>
  
  
</nav>

  
</article>

<!-- Table of Contents -->

  <aside id="toc-sidebar">
    <div id="toc" class="toc-article">
    <strong class="toc-title">Contents</strong>
    
        <ol class="nav"><li class="nav-item nav-level-3"><a class="nav-link" href="#Parameter-list"><span class="nav-number">1.</span> <span class="nav-text">Parameter list</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#Advanced-parameter-for-developers"><span class="nav-number">1.1.</span> <span class="nav-text">Advanced parameter (for developers)</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#NOTE"><span class="nav-number">1.2.</span> <span class="nav-text">NOTE:</span></a></li></ol></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Template-command"><span class="nav-number">2.</span> <span class="nav-text">Template command</span></a><ol class="nav-child"><li class="nav-item nav-level-5"><a class="nav-link" href="#Case-of-knock-out"><span class="nav-number">2.0.1.</span> <span class="nav-text">Case of knock-out</span></a></li><li class="nav-item nav-level-5"><a class="nav-link" href="#Case-of-homology-based-knock-in"><span class="nav-number">2.0.2.</span> <span class="nav-text">Case of homology-based knock-in</span></a></li><li class="nav-item nav-level-5"><a class="nav-link" href="#Case-of-Prime-Editing-Substitution-x2F-Deletion-editing"><span class="nav-number">2.0.3.</span> <span class="nav-text">Case of Prime Editing (Substitution&#x2F;Deletion editing)</span></a></li><li class="nav-item nav-level-5"><a class="nav-link" href="#Case-of-Prime-Editing-Insertion-editing"><span class="nav-number">2.0.4.</span> <span class="nav-text">Case of Prime Editing (Insertion editing)</span></a></li></ol></li></ol></li><li class="nav-item nav-level-3"><a class="nav-link" href="#The-example-of-the-command"><span class="nav-number">3.</span> <span class="nav-text">The example of the command</span></a><ol class="nav-child"><li class="nav-item nav-level-5"><a class="nav-link" href="#Case-of-knock-out-1"><span class="nav-number">3.0.1.</span> <span class="nav-text">Case of knock-out</span></a></li><li class="nav-item nav-level-5"><a class="nav-link" href="#Case-of-homology-based-knock-in-1"><span class="nav-number">3.0.2.</span> <span class="nav-text">Case of homology-based knock-in</span></a></li><li class="nav-item nav-level-5"><a class="nav-link" href="#Case-of-untreated-sample-for-negative-control-Optional"><span class="nav-number">3.0.3.</span> <span class="nav-text">Case of untreated sample for negative control (Optional)</span></a></li><li class="nav-item nav-level-5"><a class="nav-link" href="#Case-of-homology-based-knock-in-that-is-the-different-condition-from-the-above-method-Optional"><span class="nav-number">3.0.4.</span> <span class="nav-text">Case of homology-based knock-in that is the different condition from the above method (Optional)</span></a></li></ol></li></ol></li></ol>
    
    </div>
  </aside>

</section>
        
      </div>
      
      <footer id="footer">
  

  <div class="container">
      	<div class="row">
	      <p> Powered by <a href="http://hexo.io/" target="_blank">Hexo</a> and <a href="https://github.com/iTimeTraveler/hexo-theme-hiker" target="_blank">Hexo-theme-hiker</a> </p>
	      <p id="copyRightEn">Copyright &copy; 2022 - 2022 MaChIAto All Rights Reserved.</p>
	      
	      
    		<p class="busuanzi_uv">
				UV : <span id="busuanzi_value_site_uv"></span> |  
				PV : <span id="busuanzi_value_site_pv"></span>
		    </p>
  		   
		</div>

		
  </div>
</footer>


<!-- min height -->

<script>
    var wrapdiv = document.getElementById("wrap");
    var contentdiv = document.getElementById("content");
    var allheader = document.getElementById("allheader");

    wrapdiv.style.minHeight = document.body.offsetHeight + "px";
    if (allheader != null) {
      contentdiv.style.minHeight = document.body.offsetHeight - allheader.offsetHeight - document.getElementById("footer").offsetHeight + "px";
    } else {
      contentdiv.style.minHeight = document.body.offsetHeight - document.getElementById("footer").offsetHeight + "px";
    }
</script>
    </div>
    <!-- <nav id="mobile-nav">
  
    <a href="/" class="mobile-nav-link">home</a>
  
    <a href="/archives" class="mobile-nav-link">archives</a>
  
    <a href="/download" class="mobile-nav-link">download</a>
  
    <a href="/citation" class="mobile-nav-link">Citation</a>
  
    <a href="/about" class="mobile-nav-link">about</a>
  
    <a href="/contactus" class="mobile-nav-link">contact</a>
  
</nav> -->
    

<!-- mathjax config similar to math.stackexchange -->

<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    tex2jax: {
      inlineMath: [ ['$','$'], ["\\(","\\)"] ],
      processEscapes: true
    }
  });
</script>

<script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      tex2jax: {
        skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code']
      }
    });
</script>

<script type="text/x-mathjax-config">
    MathJax.Hub.Queue(function() {
        var all = MathJax.Hub.getAllJax(), i;
        for(i=0; i < all.length; i += 1) {
            all[i].SourceElement().parentNode.className += ' has-jax';
        }
    });
</script>

<script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>


  
<link rel="stylesheet" href="/fancybox/jquery.fancybox.css">

  
<script src="/fancybox/jquery.fancybox.pack.js"></script>




<script src="/js/scripts.js"></script>





  
<script src="/js/dialog.js"></script>









	<div style="display: none;">
    <script src="https://s95.cnzz.com/z_stat.php?id=1260716016&web_id=1260716016" language="JavaScript"></script>
  </div>



	<script async src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js">
	</script>






  </div>

  <div class="modal fade" id="myModal" tabindex="-1" role="dialog" aria-labelledby="myModalLabel" aria-hidden="true" style="display: none;">
  <div class="modal-dialog">
    <div class="modal-content">
      <div class="modal-header">
        <h2 class="modal-title" id="myModalLabel">PAGE VIEW</h2>
      </div>
      <hr style="margin-top:0px; margin-bottom:0px; width:80%; border-top: 3px solid #000;">
      <hr style="margin-top:2px; margin-bottom:0px; width:80%; border-top: 1px solid #000;">


      <div class="modal-body">
          <div style="margin:6px;">
            <a data-toggle="collapse" data-parent="#accordion" href="#collapseOne" onclick="javascript:setFontSize();" aria-expanded="true" aria-controls="collapseOne">
              Increase Text Size
            </a>
          </div>
          <div id="collapseOne" class="panel-collapse collapse" role="tabpanel" aria-labelledby="headingOne">
          <div class="panel-body">
            On
          </div>
        </div>
      


          <div style="margin:6px;">
            <a data-toggle="collapse" data-parent="#accordion" href="#collapseTwo" onclick="javascript:setBackground();" aria-expanded="true" aria-controls="collapseTwo">
              Dark Mode Switch
            </a>
        </div>
          <div id="collapseTwo" class="panel-collapse collapse" role="tabpanel" aria-labelledby="headingTwo">
          <div class="panel-body">
            On
          </div>
        </div>

        <div>
            <a data-toggle="collapse" data-parent="#accordion" href="#collapseThree" aria-expanded="true" aria-controls="collapseThree">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Credit&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</a>
        </div>
         <div id="collapseThree" class="panel-collapse collapse" role="tabpanel" aria-labelledby="headingThree">
          <div class="panel-body">
            MaChIAto
          </div>
          <div class="panel-body">
            Copyright © 2022 Kazuki Nakamae, Ph.D., Hiroshima Univ, Japan. All Rights Reserved.
          </div>
        </div>
      </div>


      <hr style="margin-top:0px; margin-bottom:0px; width:80%; border-top: 1px solid #000;">
      <hr style="margin-top:2px; margin-bottom:0px; width:80%; border-top: 3px solid #000;">
      <div class="modal-footer">
        <button type="button" class="close" data-dismiss="modal" aria-label="Close"><span aria-hidden="true">×</span></button>
      </div>
    </div>
  </div>
</div>
  
  <a id="rocket" href="#top" class=""></a>
  <script type="text/javascript" src="/js/totop.js?v=1.0.0" async=""></script>
  
    <a id="menu-switch"><i class="fa fa-bars fa-lg"></i></a>
  
</body>
</html>